RELEASES.md

```diff
@@ -205,3 +205,93 @@ To create a minor release:
 6. Create a pull request back to master with the CHANGELOG & version changes from the latest release.
    - Remove all `R:minor` labels from the pull requests that were included in the release.
    - Do not merge the backport branch into master.
+
+
+## Release Checklist
+
+The following set of steps are performed on all releases that increment
+the _minor_ version. These steps ensure that Tendermint is well tested, stable,
+and suitable for adoption by the various diverse projects that rely on Tendermint.
+
+### Feature Freeze
+
+Ahead of any minor version release of Tendermint, the software enters a two week
+'Feature Freeze'. A feature freeze means that _no_ new features or functionality
+are added to the code base. The Tendermint team instead directs all attention to
+ensuring that the existing code is stable and reliable. Broken tests are fixed,
+flakey-tests are remedied, end-to-end test failures are thoroughly diagnosed and
+all efforts of the team are aimed at improving the quality of the code. During
+this period, the upgrade harness tests are run repeatedly and a variety of
+in-house testnets are run to ensure Tendermint functions at the scale it will
+be used by application developers and node operators.
```

Comment:
An optional terseness-related suggestion:

```suggestion
are added to the code base. During this period, the team will focus primarily 
on ensuring the code to be released is stable and reliable. This includes:

- Fixing tests that are broken or flaky.
- Root-causing end-to-end test failures and fixing any issues they reveal.
- Proactively running upgrade tests and medium- to large-scale in-house testnets.

The key goal in this period is to minimize risk for application developers and node operators who will adopt the release candidate.
```

As a side note, one interesting question here is _how_ to do meaningful upgrade testing. Unlike testnets, where we can pretty easily create a large example case on our own, many challenges of upgrading are more to do with the specifics of an application. We can and should exercise baseline upgrade functionality, but I'm not sure how we get to a real pre-production test of upgrades without "realish" data. The obvious tactic is to use golden data, but bitrot is a real hazard for its utility.

Reply:
I still am of the view that there should be a strong enough degree of separation between Tendermint and the application that we should be able to treat application logic as a black box and not be overly concerned about the internals. There are parameters we should watch out for i.e. duration of ABCI calls, interplay between RPC and ABCI and so forth but we should be clear, when it comes to upgrading, as to what is on our side of the fence and that of the applications and be scoped on the assertions that we make when it comes to upgrading.

Reply:
> many challenges of upgrading are more to do with the specifics of an application.

I think we can get very far without use of a real application. I'm wondering if you can elaborate on what type of behavior you think requires a real-world application to test so we can discuss potentially mimicking it if need be.